File: libs/shared-types/python-src/debrief/types/tools/tool_stats.py
"""Tool statistics Pydantic models for tool metadata."""

from pydantic import BaseModel, Field


class ToolStatsModel(BaseModel):
    """Statistics about a tool's development and usage."""

    sample_inputs_count: int = Field(
        default=0,
        description="Number of sample input files available for this tool",
        ge=0
    )
    git_commits_count: int = Field(
        default=0,
        description="Number of git commits in this tool's development history",
        ge=0
    )
    source_code_length: int = Field(
        default=0,
        description="Length of the tool's source code in characters",
        ge=0
    )

    class Config:
        extra = "forbid"

File: libs/shared-types/src/zod/tools/tool_stats.ts
import { z } from "zod"

export default z.object({ "sample_inputs_count": z.number().int().gte(0).describe("Number of sample input files available for this tool").default(0), "git_commits_count": z.number().int().gte(0).describe("Number of git commits in this tool's development history").default(0), "source_code_length": z.number().int().gte(0).describe("Length of the tool's source code in characters").default(0) }).strict().describe("Statistics about a tool's development and usage.")
